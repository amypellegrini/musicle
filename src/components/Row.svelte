<script>
  import TimeSignature from "./TimeSignature.svelte";
</script>

<div class="row-wrapper">
  <TimeSignature />
  <hr class="row-line line" />
  <hr class="barline line" />
</div>

<style>
  .row-wrapper {
    position: relative;
    padding: 0;
  }

  .line {
    border-bottom: none;
    border-top-width: 3px;
    border-color: #999;
    margin: 0;
    padding: 0;
    border-style: solid;
  }

  .row-line {
    position: absolute;
    top: 24px;
    z-index: -1;
    width: 100%;
    border-bottom: none;
    border-top-width: 3px;
    border-color: #999;
  }

  .barline {
    left: 281px;
    transform: rotate(90deg);
    border-style: solid;
    position: absolute;
    width: 48px;
    top: 24px;
    border-top-width: 1px;
  }
</style>