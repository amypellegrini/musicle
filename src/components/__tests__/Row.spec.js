import Row from "../Row.svelte";
import { render, screen } from "@testing-library/svelte";

describe("Row", () => {
  it("renders", () => {
    expect(render(Row).container).toMatchSnapshot();
  });

  it("displays a time signature", () => {
    render(Row);
    expect(screen.getByTestId("TimeSignature"));
  });
});
