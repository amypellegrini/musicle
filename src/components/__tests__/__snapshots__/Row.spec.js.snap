// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders 1`] = `
<body>
  <div>
    <div
      class="row-wrapper svelte-i184u"
    >
      <svg
        baseProfile="tiny"
        data-testid="TimeSignature"
        height="49.5px"
        style="display: block;"
        version="1.2"
        viewBox="0 0 43 99"
        width="21.5px"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        
      
        <path
          class=""
          d="M3.59004,12.65 L20.77,12.65 L20.77,20.07 L12.54,20.07 C11.7601,20.07 11.1301,20.7 11.1301,21.48 L11.1301,23.39 C11.1301,24.17 11.7601,24.8 12.54,24.8 L42.79,24.8 C43.57,24.8 44.2,24.17 44.2,23.39 L44.2,21.48 C44.2,20.7 43.57,20.07 42.79,20.07 L33.85,20.07 L33.85,12.65 L42.79,12.65 C43.57,12.65 44.2,12.02 44.2,11.24 L44.2,9.32998 C44.2,8.55 43.57,7.92002 42.79,7.92002 L33.85,7.92002 L33.85,-8.04004 C33.85,-8.57998 33.55,-9.05996 33.07,-9.3 L31.8601,-9.9 C31.3301,-10.16 30.7,-10.07 30.27,-9.67002 L21.22,-1.35 C20.9301,-1.07998 20.7601,-0.709961 20.7601,-0.309961 L20.7601,7.92002 L9.25,7.92002 C9.25,7.92002 20.37,-4.10996 32.2101,-19.88 C33.0801,-21.03 32.5801,-22.08 32.39,-22.27 L30.22,-24.48 C29.95,-24.74 29.59,-24.89 29.22,-24.89 C28.62,-24.89 15.07,-24.87 14.35,-24.89 C13.6301,-24.91 13.09,-24.29 13.02,-23.67 C13.02,-23.67 12.4801,-16.31 9.88008,-8.17998 C7.28008,-0.0599609 4.34004,4.92002 1.68008,8.55 C1.68008,8.55 1.14004,9.54004 1.5,10.28 C1.85,11.03 2.48008,12.12 2.48008,12.12 C2.48008,12.12 2.74004,12.65 3.59004,12.65"
          transform="matrix(0.992,0,0,0.992,-1.29099,24.9484)"
        />
        
      
        <path
          class=""
          d="M3.59004,12.65 L20.77,12.65 L20.77,20.07 L12.54,20.07 C11.7601,20.07 11.1301,20.7 11.1301,21.48 L11.1301,23.39 C11.1301,24.17 11.7601,24.8 12.54,24.8 L42.79,24.8 C43.57,24.8 44.2,24.17 44.2,23.39 L44.2,21.48 C44.2,20.7 43.57,20.07 42.79,20.07 L33.85,20.07 L33.85,12.65 L42.79,12.65 C43.57,12.65 44.2,12.02 44.2,11.24 L44.2,9.32998 C44.2,8.55 43.57,7.92002 42.79,7.92002 L33.85,7.92002 L33.85,-8.04004 C33.85,-8.57998 33.55,-9.05996 33.07,-9.3 L31.8601,-9.9 C31.3301,-10.16 30.7,-10.07 30.27,-9.67002 L21.22,-1.35 C20.9301,-1.07998 20.7601,-0.709961 20.7601,-0.309961 L20.7601,7.92002 L9.25,7.92002 C9.25,7.92002 20.37,-4.10996 32.2101,-19.88 C33.0801,-21.03 32.5801,-22.08 32.39,-22.27 L30.22,-24.48 C29.95,-24.74 29.59,-24.89 29.22,-24.89 C28.62,-24.89 15.07,-24.87 14.35,-24.89 C13.6301,-24.91 13.09,-24.29 13.02,-23.67 C13.02,-23.67 12.4801,-16.31 9.88008,-8.17998 C7.28008,-0.0599609 4.34004,4.92002 1.68008,8.55 C1.68008,8.55 1.14004,9.54004 1.5,10.28 C1.85,11.03 2.48008,12.12 2.48008,12.12 C2.48008,12.12 2.74004,12.65 3.59004,12.65"
          transform="matrix(0.992,0,0,0.992,-1.29099,74.2849)"
        />
        
    
      </svg>
      

      
       
      <hr
        class="row-line line svelte-i184u"
      />
       
      <hr
        class="barline line svelte-i184u"
      />
    </div>
  </div>
</body>
`;
